### PR TITLE
feat: add support SDK debug mode

### PR DESCRIPTION
**Pull request by yeul**

### CODE REVIEW

File: src/components/message/Message.jsx [hunk 1]
```diff
@@ -99,6 +100,8 @@ const Message = () => {
         });
     }, [amount, currency, buyerCountry, JSON.stringify(style), offer, payerId, clientId, merchantId]);
 
+    ppDebug(`Correlation Id: ${meta.ppDebugId}`);
```
Comment (gillycheesesteak) [inline src/components/message/Message.jsx]: Let's put this in a useEffect or maybe in the onReady callback. Also, we may want to add the same type of log to the Modal requests as well

File: src/controllers/message/debug.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    if (/(\?|&)pp_debug=true(&|$)/.test(window.location.search)) {
+        if (debugObj) {
+            // eslint-disable-next-line no-console
+            console.warn('PayPal Messages Debug:', message, debugObj);
+        } else {
+            // eslint-disable-next-line no-console
+            console.warn('PayPal Messages Debug:', message);
+        }
+    }
+}
```
Comment (gillycheesesteak) [inline src/controllers/message/debug.js]: Would this make sense in a new or existing `utils/` file?

File: src/controllers/message/interface.js [hunk 3]
```diff
@@ -125,6 +126,25 @@ export default (options = {}) => ({
...
+                    ignoreCache: ${ignoreCache},
+            
+                    index: data-pp-id="${index}"
+                    style: ${JSON.stringify(style)},
+                    amount: ${amount},
+                    buyerCountry: ${buyerCountry},
+                    placement: ${placement},
+            
+                    renderStart: ${renderStart},
+                    renderMessageTime: ${new Date().getTime()}
```
Comment (gillycheesesteak) [inline src/controllers/message/interface.js]: Would this be more helpful as a date/timestamp?

File: src/controllers/message/debug.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+/**
+ * Messaging debug log. When the query param "pp_debug" is used on a page with a message and set to "true", a series of logs will appear
+ * within the browser's developer tools console.
+ * @param {string} message The name of the parameter you are logging out. Also often times includes the parameter to be logged
+ * within a template string.
+ * @param {object} debugObj Optional argument used in cases where the property you wish to log is an object. i.e.
+ * an HTML element.
+ */
+export function ppDebug(message, debugObj) {
+    if (/(\?|&)pp_debug=true(&|$)/.test(window.location.search)) {
```
Comment (gillycheesesteak) [inline src/controllers/message/debug.js]: This check is not working currently inside the component iframes

File: src/controllers/message/setup.js [hunk 5]
```diff
@@ -7,10 +7,11 @@ import {
     getAccount,
     getCurrency,
     getPartnerAccount,
-    getInsertionObserver
+    getInsertionObserver,
+    formattedDate
 } from '../../utils';
 import Messages from './adapter';
-import { ppDebug } from './debug';
+import { ppDebug } from '../../utils/debug';
```
Comment (gillycheesesteak) [inline src/controllers/message/setup.js]: This can be added to other `../../utils` imports

Comment (yeul) [reply src/controllers/message/setup.js]: Oh duh, I totally missed that. Good catch, thank you.

File: src/controllers/message/debug.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
+/**
```
Comment (gillycheesesteak) [inline src/controllers/message/debug.js]: I think this file snuck back in somehow

File: src/utils/debug.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+ * within the browser's developer tools console.
+ * @param {string} message - The name of the parameter you are logging out. Also often times includes the parameter to be logged
+ * within a template string.
+ * @param {object} options - Allows for an optional debugObj and/or an optional inZoid component check argument.
+ * @param {object} [options.debugObj] - Optional argument used in cases where the property you wish to log is an object. i.e.
+ * an HTML element. ppDebug('Script:', { debugObj: script }).
+ * @param {boolean} [options.inZoid] - Optional argument. Set inZoid to true if you wish to log inside of the iframe. i.e. ppDebug('Sample message', {inZoid: true}). Not using this
+ * argument defaults inZoid to false.
+ */
+export function ppDebug(message, options) {
```
Comment (gillycheesesteak) [inline src/utils/debug.js]: consider this to make the options "self-documenting"
```suggestion
export function ppDebug(message, { inZoid, debugObj } = {}) {
```

File: src/utils/debug.js [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+    const inZoid = options?.inZoid;
+    const debugObj = options?.debugObj;
+    if (/(\?|&)pp_debug=true(&|$)/.test(window.location.search) && !!inZoid === isZoidComponent()) {
+        if (debugObj) {
+            // eslint-disable-next-line no-console
+            console.warn('PayPal Messages Debug:', message, debugObj);
+        } else {
+            // eslint-disable-next-line no-console
+            console.warn('PayPal Messages Debug:', message);
```
Comment (gillycheesesteak) [inline src/utils/debug.js]: Can this be combined into one line but maintain the desired output?
```suggestion
            console.warn('PayPal Messages Debug:', message, debugObj ?? '');
```

File: src/controllers/message/debug.js (hunk 6 above)
Comment (yeul) [reply src/controllers/message/debug.js]: Oops! Fixed now.

**Review by gillycheesesteak — APPROVED** [review]

**Pull request merged**